### ISSUE TITLE
Klipper logo

### ISSUE THREAD

**Issue by BKLronin** [KevinOConnor/klipper#822]
Is there any logo already for the firmware? I could possibly make a simple one.

**Comment by mamoit** [comment]
This would be awesome, I want to put a sticker on my Orange pi case, I just need something to put on the sticker :smile:

**Comment by Hywelmartin** [comment]
i'm not sure if "Klipper" is some what associated with Clipper(boat type).. but I found this..
![grey-ink-sailing-ship-leg-tattoos-1](https://user-images.githubusercontent.com/33714214/47572337-fe257b80-d93a-11e8-811a-a97a4504ddbe.jpg)

**Comment by BKLronin** [comment]
I initially thought about a simple word logo that expresses the heart of klipper with its lines directly. A klipper as minimal graphic would be nice too. Depends on what Kevin might like more. @KevinOConnor

**Comment by KevinOConnor** [comment]
FYI, @shoreliner proposed this a few months back:
![artboard 1mdpi](https://user-images.githubusercontent.com/3004890/47605114-2d39fc80-d9d0-11e8-99fc-8c702be97311.png)

**Comment by theopensourcerer** [comment]
I prefer the circuit board idea to the ship, but both seem overly complex for a logo.

@kevin why is Klipper called Klipper?

**Comment by BKLronin** [comment]
> FYI, @shoreliner proposed this a few months back:
> ![artboard 1mdpi](https://user-images.githubusercontent.com/3004890/47605114-2d39fc80-d9d0-11e8-99fc-8c702be97311.png)

Wow, that looks excellent, I would vote for that. I don´t think it will get any better. (I can´t for sure).

**Comment by frankvdh** [comment]
I like the idea of the clipper ship and the octopus, linking to Octopi. But
I think a logo needs to be simpler than the above examples, so I did a
search for kraken and clipper, and found the attached, which I think would
be a good starting point...

**Comment by bruce356** [comment]
Hi, I think the K looks good but with out the spaghetti and without the second examples white dots surrounding it.
regards - bruce

**Comment by LazaroFilm** [comment]
I was also working on a logo on my spare time. Definitely not finished but since the conversation is going... I was also for the Klipper Ship and Kraken tentacles (note, I did this logo before seeing any of this conversation) 

![klipper](https://user-images.githubusercontent.com/18351100/47620752-445a1680-dac4-11e8-9af4-ea78c65dad7b.png)

**Comment by KevinOConnor** [comment]
> @KevinOConnor why is Klipper called Klipper?

There is no particular reason for it.  I wanted a name that would be distinct on a web search.  I had a few different names I was thinking about (none of which I now remember) and I just choose one.

-Kevin

**Comment by BKLronin** [comment]
I had somethin clear and edgy in mind. Trying to express precision and fine lines, with a light 80ies touch. 
Something like this loose sketch.
![klipperlogosketch](https://user-images.githubusercontent.com/6392076/47622875-8443ec80-db0a-11e8-9ff0-32b423e1982d.png)

**Comment by KevinOConnor** [comment]
Interesting.  I don't have any drawing abilities so I don't have much comment to give on a logo.  My personal preference of the above would be the "K" logo.  Though I agree that the wires outside the "chip" in that logo could be removed.

-Kevin

**Comment by aeropic** [comment]
![k](https://user-images.githubusercontent.com/38628543/47816880-8c8f6800-dd54-11e8-994f-5fce78bd43cf.jpg)
What about something like that with the K symbolizing the hot end and some filament extruded from it ?
I must admit I'm not an expert in drawing ;-)

**Comment by aeropic** [comment]
![klip](https://user-images.githubusercontent.com/38628543/47820032-87361b80-dd5c-11e8-9880-21b50d340a75.png)
Or this one ?

**Comment by aeropic** [comment]
Pushing my idea of the K hot end... I propose this one:
![klipper](https://user-images.githubusercontent.com/38628543/47849363-95724f00-ddd1-11e8-832b-9ce83ede9adb.png)

**Comment by theopensourcerer** [comment]
FWIW, I like the idea of using the K like a hot-end/extruder. I am wondering though, for a Logo, do you really need to have the "ipper" as well?

I am thinking of, for example, the logo for KDE, e.g.

https://www.kde.org/stuff/clipart/logo/logo-white-blue-source.svg

**Comment by aeropic** [comment]
@theopensourcerer , a minimal version for you !
![minimal](https://user-images.githubusercontent.com/38628543/47849536-4f69bb00-ddd2-11e8-84bb-20b02dfaa247.png)

I like it as well as a logo it's maybe enough ;-)

**Comment by theopensourcerer** [comment]
I think we need more feedback and input from others but I for one certainly like the idea.

**Comment by BKLronin** [comment]
I´m still a fan of the reduced shoreliner version but aeropics idea is essentially nice as well. 
I minified his version a llittle more. Needs finalizing for sure.
![minified](https://user-images.githubusercontent.com/6392076/47851477-0bc67f80-ddd9-11e8-91f5-9fd7535a05ab.PNG)

**Comment by mgth** [comment]
I Love the idea of the k hotend, I think the L is too much, maybe something more graphical
![klipper_logo_study](https://user-images.githubusercontent.com/2568124/47851807-06b60000-ddda-11e8-8535-bed1c2789251.png)

**Comment by aeropic** [comment]
@mgth 
You're right the L is too big.
Here is a reduced version
![klipper_small_l](https://user-images.githubusercontent.com/38628543/47852218-62cd5400-dddb-11e8-9589-987054c2b5c6.png)

**Comment by aeropic** [comment]
@BKLronin , your hot end is flying too high, you'll get poor adhesion of the first layer ;-)

**Comment by Avalonnw** [comment]
May I suggest just k-hotend without L?

**Comment by aeropic** [comment]
@Avalonnw  Here it is ... Maybe somewhat too minimalist ?
![k alone](https://user-images.githubusercontent.com/38628543/47857193-4d5f2680-dde9-11e8-82f7-9d7d68addba7.png)

**Comment by Avalonnw** [comment]
Good logo should be readable in monochrome. 2 colors are good but if you
convert red to blsck it will be unreadable.

**Comment by blaskkaffe** [comment]
I have not had time to try my ideas yet but I think it would be nice to somehow show the separation between processing/scheduling and doing the mechanical work which is the bit thing that makes klipper klipper.

One thought I had was half a brain, half cog/hotend/stepper motor. Alternatively if you guys liked the boat idea a thougt was to have a lighthouse/captain/map and a boat for example. I have done some sketches but will try to make something worth showing :p

**Comment by aeropic** [comment]
@Avalonnw , you're right
The black & white or grayscale issue is easy to solve.
Either inserting one pixel white around the letter L or painting it into another less saturated color.
Or both !

![klipper_gray](https://user-images.githubusercontent.com/38628543/47899549-2a308780-de7a-11e8-9a7c-eee01b3cf16b.png)

**Comment by aeropic** [comment]
we could even play with Kevin's initials ;-)
![koc](https://user-images.githubusercontent.com/38628543/47899839-218c8100-de7b-11e8-92d6-6f66ed2814e8.png)

**Comment by BKLronin** [comment]
Next iteration with some of the ideas integrated.
![k_lipper](https://user-images.githubusercontent.com/6392076/47906037-ae8a0700-de88-11e8-93c3-0a32670fd715.png)

**Comment by Avalonnw** [comment]
Overloaded with small details IMHO

**Comment by BKLronin** [comment]
Yes, I had the same impression. Still trying to find out how to express klippers nature with few elements

**Comment by dragonnn** [comment]
The last one without the squares would be the best.

**Comment by BKLronin** [comment]
I like the squares, expressing the companion idea between raspi and 8 bit but it would like this. 
![minimal](https://user-images.githubusercontent.com/6392076/47906932-88656680-de8a-11e8-8027-31ee98cbb6b7.PNG)

**Comment by aeropic** [comment]
Not bad... why not enlarging the "Lipper" to something like 
![k bigger](https://user-images.githubusercontent.com/38628543/47907117-e72ae000-de8a-11e8-8b7a-ae0b98d69bf1.png)

**Comment by BKLronin** [comment]
Like this? 
![enlarged](https://user-images.githubusercontent.com/6392076/47907375-6c15f980-de8b-11e8-9783-1b9ed10791d0.PNG)

**Comment by aeropic** [comment]
Yes, left one...
Now, just do one witthout the red squares, enlarge the bottom part of the big L so that it fits the width of the other letters, enlarge the "LIPPER" to fill the square and just try to make the parts of the letters PER overlapping the K in white...

**Comment by aeropic** [comment]
Something like that ... but with the overlapping part of the letters PER in white :
![klogo](https://user-images.githubusercontent.com/38628543/47908716-18f17600-de8e-11e8-9d7b-2401eadd8377.jpg)

**Comment by aeropic** [comment]
With Arial rounded bold police
![k_arial_rounded](https://user-images.githubusercontent.com/38628543/47909299-c6b15480-de8f-11e8-9442-1da99856cbef.png)

**Comment by BKLronin** [comment]
Like this
![cutlarger](https://user-images.githubusercontent.com/6392076/47909438-2e679f80-de90-11e8-975d-d732f0e3938b.PNG)

**Comment by aeropic** [comment]
It's becoming very good !
Please just try to find a police that is higher and maybe thinner in width to be more balanced with the tall L ... See above the Arial Rounded bold ;-)

**Comment by BKLronin** [comment]
I lik ethe font so far but here some alternatives 
![snapshotfont](https://user-images.githubusercontent.com/6392076/47912020-73430480-de97-11e8-9a77-990550ba3aec.PNG)

**Comment by aeropic** [comment]
I love the top right one. We can see the deposited filament flowing out of the nozle !

**Comment by BKLronin** [comment]
> I liked the top right one too. But, I think my preference would be to remove the part of the "K" that is behind the "per".

Do you mean the "only letter cutout" or the "bigger square cutout"?

**Comment by BKLronin** [comment]
![cutfont](https://user-images.githubusercontent.com/6392076/47913220-940d5900-de9b-11e8-9390-2c3d227e2c98.PNG)

**Comment by theopensourcerer** [comment]
Yes, that.

**Comment by theopensourcerer** [comment]
The more I look at that the less I like it  - lol... Perhaps then the one you also did where the "per" text became transparent in the bottom of the K - but with this typeface? Maybe that would look better?

**Comment by BKLronin** [comment]
![cutoutvari](https://user-images.githubusercontent.com/6392076/47913742-7a6d1100-de9d-11e8-9da9-ca4977d9fef7.PNG)

**Comment by aeropic** [comment]
Agreed, I don't like neither the cropped nozzle...
Here is another attempt with the overlapping text made white and a filled nozzle so that it becomes more obvious it's a nozzle but the K is still present...
![fullk](https://user-images.githubusercontent.com/38628543/47913740-79d47a80-de9d-11e8-8d3d-1cd573500f0c.png)

**Comment by BKLronin** [comment]
I had the triangle filled in an internal version but it instantly reminded me of the Kodak logo.

**Comment by theopensourcerer** [comment]
Gosh this is harder than it looks to find something that looks "right"... I'm not sure I like the transparent text or the filled in nozzle to be honest. I think something like these is very close but I feel I am not enough of a designer to be able to comment much more.

I guess what we should really do is get some input from Klipper's SABDFL ;-) 

Or several capable people come up with ideas and a vote is held...

**Comment by theopensourcerer** [comment]
FWIW - My preferred design so far is this one (although I am not sure it is "right"):

https://user-images.githubusercontent.com/6392076/47913220-940d5900-de9b-11e8-9390-2c3d227e2c98.PNG

**Comment by aeropic** [comment]
Which means we are very close to a perfect logo !!!!
A pitty Kodak uses a hot end with a non flat bed ;-)
![kodak](https://user-images.githubusercontent.com/38628543/47914078-8f966f80-de9e-11e8-9110-e9d115d5a6ee.png)

Look it works really fine in small like a PC icon
![small](https://user-images.githubusercontent.com/38628543/47914086-9624e700-de9e-11e8-9d80-e8f066db22c4.png)

**Comment by BKLronin** [comment]
We don´t have to nail it just now but I think. I pretty much like this one but I´m biased as well.  

![love](https://user-images.githubusercontent.com/6392076/47914121-b8b70000-de9e-11e8-80a3-735b84c250a5.PNG)

**Comment by mamoit** [comment]
Left one!

**Comment by aeropic** [comment]
Much nicer in red filament, I agree !!! Don't change anything and let all the others vote ;-)

In red, left one is cool !

**Comment by theopensourcerer** [comment]
Left. Looks really good to me. Well done.

**Comment by BKLronin** [comment]
Thanks, I prefer the left one as well. :)

**Comment by aeropic** [comment]
Just a try with a redFilament
![red_full](https://user-images.githubusercontent.com/38628543/47914468-126bfa00-dea0-11e8-90c4-9f367b31780c.png)
_full nozzle

**Comment by BKLronin** [comment]
A bit to heavy for me. I like the slightly "hidden" nozzle.

**Comment by mamoit** [comment]
Can you try the previous version (non-full) but connect the top of the 'L' to the red outer part of the 'K', as if the filament was going through the extruder hole.

**Comment by aeropic** [comment]
Lighter with fruit inside ;-)
![red_full_rasp](https://user-images.githubusercontent.com/38628543/47915028-da65b680-dea1-11e8-8da0-fc60a5fc8234.jpg)

**Comment by dragonnn** [comment]
bottom left first +1. Maybe we should do a polling?

**Comment by aeropic** [comment]
What's the difference between the two in the right top and bottom ? A subtle shade of gray ? (I think there are at least 50 shades ;-)
The ones I don't like are  top2, bottom2 bottom4

**Comment by BKLronin** [comment]
Nr 8 has a slight 3d effect on the K.

**Comment by BKLronin** [comment]
I don´t know how to do a poll. Lets call Kevin :D

@KevinOConnor @shoreliner

**Comment by aeropic** [comment]
I like it but we should first focus on the overall shape, then improve the rendering ! What would give n°8 and n°3 with a slight separation between gray and red (more or less what you did with the extruded filament) ?

I think poll is easy to do on facebook but not sure everyone of us has a facebook account...

**Comment by dragonnn** [comment]
Best way would be to use Google poll and put a link here

**Comment by BKLronin** [comment]
https://docs.google.com/forms/d/e/1FAIpQLSfj-bjnshuuIxN5HMtnvbkTlQ8N7jV_UCJyEtr0a1l_awQz5Q/viewform?usp=sf_link

**Comment by aeropic** [comment]
Once you've voted please add a like on the post above of BKL so that we can estimate the volume of votes !

**Comment by LazaroFilm** [comment]
I like the version 5. My only gripe is the PP are not easy to read.

**Comment by LazaroFilm** [comment]
I voted for #5

**Comment by Hywelmartin** [comment]
I couldn't help my self

#5 Logo as STL for One filament.....

[klipper_Logo_stl.zip](https://github.com/KevinOConnor/klipper/files/2543502/klipper_Logo_stl.zip)

**Comment by Hywelmartin** [comment]
Damned...

**Comment by BKLronin** [comment]
I think we should wait a bit more before the "merchandising" production begins. :D

I will post the SVG and things when the vote was open for a few days.

**Comment by BKLronin** [comment]
Couldn´t resist as well :D
![merchandising](https://user-images.githubusercontent.com/6392076/47935987-9b088b80-dedc-11e8-9c19-ab1c92944322.PNG)

**Comment by aeropic** [comment]
I'll take a XL please ;-)

**Comment by BKLronin** [comment]
> I'll take a XL please ;-)

If this gets accepted I will definetely wear one. :D

**Comment by aeropic** [comment]
I :heart: it on the gray teeshirt ... perfect match of colors !

**Comment by hg42** [comment]
this looks nice...

however... :-)
* the tail of the R (bottom right line) doesn't line up with the red inner of the nozzle...(may be these should have the same angle and may be the same line width)
* eventually the upper right line of the K could be symmetrical to the lower right one, this would further emphasize the nozzle and make it more balanced, the vertical line could eventually be shortened instead
* it doesn't work for pure black and white (so there needs to be some space between filament and nozzle)
* it's too complicated for a small version

But there could be an alternate version without "IPPER".
One only the K, very simplified and graphical, the other with color and more detailed with full title.
The K should probably be similar enough for both to recognize that they are the same.

I like the golden 3D version, also because it's turned a bit, may be the logo could be rotated between 30 and 60 degree? this would also tell the viewer more about the intention behind it (nozzle).​

**Comment by BKLronin** [comment]
> the tail of the R (bottom right line) doesn't line up with the red inner of the nozzle...(may be these should have the same angle and may be the same line width)

The inner reddish line ends at the tip of the R. I think it would break the font when that part of the R is rotated. 
> eventually the upper right line of the K could be symmetrical to the lower right one, this would further emphasize the nozzle and make it more balanced, the vertical line could eventually be shortened instead
We had the symmetrical triangle in earlier versions but dropped that part of K for a better raedability. In Version 2 there is still the cut out version present.
> it doesn't work for pure black and white (so there needs to be some space between filament and nozzle)
That space will be added as in Version 4 and 8, those are a bit more recent.

>  doesn't work for pure black and white
Thats another constraint. I found greyscale quite enough. It is recognizable when printed at the moment.

I tried rotating it but while it looks quite interesting I lost the ability to instantly read it. The K disappears a little and it looks a bit "untidy".

3D could be useful for intros eventualy or video in general. Would like to keep the logo as flat as possible while looking solid.
![logo](https://user-images.githubusercontent.com/6392076/47950232-90d3a500-df4f-11e8-96e4-76b19672df8a.png)

A minified black and white could look like this:

![minifiedbw](https://user-images.githubusercontent.com/6392076/47950286-60d8d180-df50-11e8-9c1d-8731822b82f3.PNG)

**Comment by Hywelmartin** [comment]
I altered the R, so that the cancan leg is co-linear with the plastic in the nozzle

I kind of like the controller part of the early suggestion 

for a minimal logo I think that @BKLronin :s black and white is good

![kliopper_logo_chipp_render](https://user-images.githubusercontent.com/33714214/47950762-2d4d7580-df57-11e8-994e-3299b38e27cb.PNG)

**Comment by BKLronin** [comment]
The extruded line in the minimal looks good but I think its not necessary. The chip is also too much.

**Comment by aeropic** [comment]
After long relexion, I've a concern with the red filament inside the nozzle. As the logo is flat, the top right corner of the red filament (the big "<") finishing with a vertical line gives the feeling of a 3D object then going to the left it finishes as a 2D and make a kind of optical illusion like an impossible object...
We could maybe try to end the red V at zero to the right (see the blue line) ? Or something else to brerak this 3D effect ... ?
![k-concern](https://user-images.githubusercontent.com/38628543/47951716-d9e22400-df64-11e8-9eb8-be9e5755a4da.png)

**Comment by BKLronin** [comment]
![changes](https://user-images.githubusercontent.com/6392076/47951812-59242780-df66-11e8-956a-6c77d433f4b6.PNG)

**Comment by aeropic** [comment]
Thanks,
:heart: 2nd one !!!!

**Comment by KevinOConnor** [comment]
On Fri, Nov 2, 2018 at 9:42 AM Thomas Herrmann <notifications@github.com>
wrote:

> I don´t know how to do a poll. Lets call Kevin :D
>

The recent work is very impressive.  I will only have limited internet
access for the next several days, so
I wont be able to provide much in the way of comments until I return.

Separately, I'd be curious how the logo would look if "Klipper" was spelled
out vertically along the stem
of the K in filament (using lower case letters with each letter's stem next
to the stem of the K and the "extruder" part of
the K somewhere near the lower case "i").  I suspect it would not look as
good as the current work - just
a thought.  I'll try to mock something up when I return.

-Kevin

**Comment by Hywelmartin** [comment]
-ish?
![klipper_logo_klipperk](https://user-images.githubusercontent.com/33714214/47955377-b6849c80-df97-11e8-9f34-5f41e1f3e42c.PNG)

**Comment by shoreliner** [comment]
Lol This escalated quickly. =)  I'll make you guys a trade, working SAM4S
build for the Duet Maestro for an improvement on the original design?
Lol.... j/k, you guys are doing GREAT =)  I really like the
simplifications!  I did that black/white one in a matter of 20 minutes and
didn't think of much else.  It's the Logo we use for the Discord channel on
Klipper... I'd gladly adopt the new one.

Thanks,

David Hargrove
streamliner@printM3D

**Comment by BKLronin** [comment]
I still like you logo to be honest. The only downside I may see is that it is a bit more general. The 3D printing link is missing so to speak. Other than that a great design that does look pretty good even when small. 

I would accept this as logo as well, didn´t knew it is used on discord. Didnt even know there is a channel. We could do another vote between your original and the #5 K.

**Comment by BKLronin** [comment]
Thats the current versions as SVG inside a pdf. Just open up with inkscape and degroup.
[Collection.pdf](https://github.com/KevinOConnor/klipper/files/2545319/Collection.pdf)
And the current results of the poll.
![logopoll](https://user-images.githubusercontent.com/6392076/47957060-9b258b80-dfaf-11e8-89bb-c06686f1c2e0.PNG)

**Comment by Hywelmartin** [comment]
It looks strange, I don't really know where to put the "I"
![vertikalwordcapshorisontalletter](https://user-images.githubusercontent.com/33714214/47957970-38d58680-dfc1-11e8-975d-31dfaee09b6d.PNG)
![vertikalwordcapshorisontalletter_i_to_left](https://user-images.githubusercontent.com/33714214/47957971-38d58680-dfc1-11e8-98d8-ac194b40559d.PNG)

**Comment by Hywelmartin** [comment]
@aeropic The color is less important right now. I'm doing this in 3d-cad
I started with a square so I agree. to the squariness .
I attached the filament to the closest/most simple letter just to get a feeling of the vertical word with horisontal letters....

For the emphasis on either K or nozzle I need a hint...


added some filamet... 

![klippergoldenwingslogo](https://user-images.githubusercontent.com/33714214/47962069-37cf4400-e017-11e8-8733-f4fc14486da4.PNG)

and a little more evil steel nozzle
![klippergoldenwingslogosteel](https://user-images.githubusercontent.com/33714214/47962070-3867da80-e017-11e8-9991-b196fb2d1ea5.PNG)

**Comment by aeropic** [comment]
I :heartbeat: the one with the dot on the I !

**Comment by Hywelmartin** [comment]
An alternative for the K is to use the same font
I'm thinking of centralize the uotput from the nozzle, but I'm spending too much time on this..... but its fun...
![klipperklogo](https://user-images.githubusercontent.com/33714214/47963526-d49cdc00-e02d-11e8-9671-8fb06628edff.PNG)

**Comment by aeropic** [comment]
Humm, I do not like this weird hotend :unamused:

**Comment by LazaroFilm** [comment]
Yeah not a fan of the weird nozzle either. I do like the previous one with the dot on the i.

**Comment by Hywelmartin** [comment]
I have read thru the latest posts and conclude it to this

#1 
![snedlogo](https://user-images.githubusercontent.com/33714214/48003220-1e5cf380-e10e-11e8-9bbb-44a3fbeebf67.PNG)
the suggestion to cut of the right part of the K and move it makes it really crocked, and looks upside down i'm not sure

#2  
![intebralogo](https://user-images.githubusercontent.com/33714214/48003380-9a573b80-e10e-11e8-91c2-72449969fd32.PNG)
A K that follows the font of lipper was no good

#3  
![bralogo](https://user-images.githubusercontent.com/33714214/48003417-b0fd9280-e10e-11e8-9da3-48ba8069e9aa.PNG)
this one was good

#4 
![braenkel](https://user-images.githubusercontent.com/33714214/48003432-bce95480-e10e-11e8-8156-d0b71fda5847.PNG)
This one was also good

#5 
![logochip](https://user-images.githubusercontent.com/33714214/48003555-12256600-e10f-11e8-89e6-7ed0a7e81726.png)
This one was good

#6 ehhm...#5 originally
![ocksabra](https://user-images.githubusercontent.com/33714214/48003682-6a5c6800-e10f-11e8-9f76-50e09796fb67.PNG)
Good as well

So how go forward maybe @KevinOConnor should have a say??

**Comment by Papsnoek** [comment]
This is my favorite that is not in the post above:

![image](https://user-images.githubusercontent.com/3797281/48004094-ec9d5a00-e118-11e8-8f90-423df1b8b519.png)

**Comment by Hywelmartin** [comment]
@Papsnoek   sorry... it's there now

**Comment by BKLronin** [comment]
I like this one the most.  The Vertical Klipper could be turned 90 degree as well. Otherwise there are two Ks in there. The gold only works for renderings and turns to simple yellow on a normal vector. Best vote would be between shoreline K and the original No5.

![tshirt](https://user-images.githubusercontent.com/6392076/48005675-4ea79080-e114-11e8-923c-3df48e870815.png)

**Comment by LazaroFilm** [comment]
I like the #7, but on the side. 
![48004462-8234eb80-e111-11e8-94c6-cfff73483abd](https://user-images.githubusercontent.com/18351100/48019823-5cecb100-e102-11e8-97aa-667f509af9e9.png)

**Comment by mamoit** [comment]
I have to say that I prefer the lettering on Number 3, if it could be on its side and in red it could look even nicer than this one :point_up:

**Comment by theopensourcerer** [comment]
I still favour the #5 in @BKLronin 's post:

https://github.com/KevinOConnor/klipper/issues/822#issuecomment-435379113

**Comment by wizhippo** [comment]
I know klipper at the moment is 3D printer oriented but it could be modified for other nc machines such as lazer cutters etc.  Perhaps an extruded isn’t the best thing to add to the logo if this was to happen.  Don’t hate:)

**Comment by mgth** [comment]
![text1048-2](https://user-images.githubusercontent.com/2568124/48034419-a4555a80-e15f-11e8-9ee7-080d9c7f4b58.png)

**Comment by mgth** [comment]
![path852-6-3](https://user-images.githubusercontent.com/2568124/48049177-1f3d6600-e19e-11e8-8376-3aa4ce4c1907.png)
Just for fun

**Comment by mgth** [comment]
![path1396-9-0-8-76](https://user-images.githubusercontent.com/2568124/48060621-c7fabe00-e1bc-11e8-96c3-2e30b02e4ba6.png)
Because it is python cnc controller...

**Comment by hg42** [comment]
@wizhippo
> I know klipper at the moment is 3D printer oriented but it could be modified for other nc machines such as lazer cutters etc. Perhaps an extruded isn’t the best thing to add to the logo

I agree...

Klipper can do quite a lot of things, because it's modular (and I guess it will be become even more modular in the future).
Basically it's a high level host combined with several MCUs.
What is the main feature? perhaps:
"Calculating an exactly timed schedule of actions and executing it elsewhere"
Currently I don't have a clue, how to express that graphically...

**Comment by mgth** [comment]
![text1048-2-69-78](https://user-images.githubusercontent.com/2568124/48149146-b1875c00-e2bb-11e8-826f-6d848e329de1.png)
here we have motion, timing, python, could not figure how to talk about MCUs...

**Comment by shoreliner** [comment]
Apologies guys -- work takes me away from Github most of the time these days.  I still like everything here; I am glad to see I wasn't the only one trying hard to figure out how to describe Klipper in an image. =)   If you do want my original logo for anything, please let me know and I'll share it in SVG/AI/etc formats.  I have to agree with @Hywelmartin for the most part; that breakdown is a good take on my opinions of what has been produced here.  Maybe consider a couple of versions of the logo similar to the first you have listed.  A more simplified "K" of some sort, and then the stylized logo that says "Klipper" as is listed.  Just a few thoughts!  Cheers!

**Comment by BKLronin** [comment]
@shoreliner 
Could you please attach the svg? Easier to play with.

**Comment by mgth** [comment]
![path14203](https://user-images.githubusercontent.com/2568124/48296136-688a0000-e493-11e8-8089-5da9add71450.png)

**Comment by BKLronin** [comment]
Still thinking about this but here is a minimal logo base on my previous ones. I like the color combination a lot. I find it difficult to get all the suggestions into it without making it overloaded. Maybe thats not necessary. I took the speed topic into this one alone. It´s a Skip symbol as K. Meaning fast forward or I just can´t wait to get this part finished :D

![speedklipper](https://user-images.githubusercontent.com/6392076/48300989-a9f7cb00-e4e6-11e8-9b23-0b915f50fd7f.PNG)

**Comment by aeropic** [comment]
@BKLronin ,
I like this last one "fast forward" a lot as I can see:
- the fast forward symbol,
- the thinking out of the box of Klipper
- the extruder
- KLIPPER
and it is very neat and simple . No need to try to put everything into a logo, the best logos (nike, apple...) don't say too much about the brand itself except one single idea !

You could maybe try to open the the noozle and separate slightly the K into two parts : ">| " ?

**Comment by aeropic** [comment]
Agreed, keep this one ! At least  the people who want a generic logo (CNC Klipper) can see only a fast forward K ! While we, 3D printers, can see a fast moving K-hot End ;-)

==> BRILLIANT IDEA :+1:

**Comment by aeropic** [comment]
PS : print me a Tee shirt please :rofl:

**Comment by BKLronin** [comment]
Thanks :)m Here you go-> :D
![klippershirt](https://user-images.githubusercontent.com/6392076/48305337-33c88800-e529-11e8-8587-284efcb8e268.PNG)

**Comment by KevinOConnor** [comment]
FYI, this is roughly what I was thinking of in my [previous message](https://github.com/KevinOConnor/klipper/issues/822#issuecomment-435593615).
![klipper](https://user-images.githubusercontent.com/3004890/48319066-f3274800-e5d6-11e8-8dc7-f437d50eea8c.png)
I don't think the idea is as good as some of the other ideas posted, but I figured I'd post it in case it inspired any other ideas.

Just my 2 cents, but I like the image:
![](https://user-images.githubusercontent.com/6392076/48005675-4ea79080-e114-11e8-923c-3df48e870815.png)
and the similar:
![](https://user-images.githubusercontent.com/33714214/48003682-6a5c6800-e10f-11e8-9f76-50e09796fb67.PNG)
the most so far.

-Kevin

**Comment by BKLronin** [comment]
Thanks for the Feedback. Kevin

I had a similar thought when placed on the T-shirt that the "big-K" looked pretty good. Thats why I´m putting it on the template to see how it reacts.

I will try the vertical approach soon.

**Comment by BKLronin** [comment]
Good point, thats often a bit of a problem. I think the size of the Logo here on the page makes it more difficult to read. Most of the time it will be way smaller and more readable through that, as you read the K first. Trying to make some  new drafts soon with the ideas.

**Comment by DkOuvE** [comment]
I finally got around to making my own concepts from the 2 images in the top right. I was going for a "Faster than the rest" idea with the first one (the l and the i look like bar charts), and "Fast Extrusion" feel with the second one.
![image](https://user-images.githubusercontent.com/23626715/50043938-9b1baf80-0074-11e9-98aa-12df3e2d90af.png)

**Comment by hg42** [comment]
> ![klipper-logo](https://user-images.githubusercontent.com/4707597/50745106-a9e72380-1228-11e9-9a2f-08a0ac91dcaa.png)

I like the KISS approach and the reversed "K". Also the font is very nice in general and it looks very much like filament paths (better than the other fonts used above, e.g. no sharp corners).

> ![klipper-logo-small](https://user-images.githubusercontent.com/4707597/50745110-b3708b80-1228-11e9-92f9-cdc6fa67a247.png)

I think the K alone (reversed and with a nozzle channel) is recognizable enough for a space limited logo.

I guess, both "K"s should probably be the same.

I would like to see the font of the other letters to be more similar to the single K, so more rounded at the ends and the line width consistent at all places.
I think it is possible to retrace the font with some simple paths.

here are some variations of your design:

![red-gray svg](https://user-images.githubusercontent.com/4707597/50748234-a491d500-1238-11e9-8af8-ceab99b69d50.png)

I personally like the second and the last

**Comment by hg42** [comment]
and then there is _KliPPer_ (a sketch on a beer coaster):

![klipper](https://user-images.githubusercontent.com/4707597/50750668-3c49f000-1246-11e9-8e52-5b3f9c9cae9d.png)

**Comment by DruckiMcDruck** [comment]
Most of the proposals look like *lipper* with an awkward shape right next to it, sorry. 

Someone posted this 3D print in a facebook group and i really liked it. simple, readable and (obviously) 3d printable.
![grafik](https://user-images.githubusercontent.com/47270566/52164917-6aadd480-26f9-11e9-9702-88d6dc60ffad.png)

**Comment by BKLronin** [comment]
> I backed the K of a little and corrected the distance to the L. While it looks nice I think that the nozzle tunnel breaks the Ks readability a little. So I would like to leave this to viewers imagination if this i a nozzle.
> 
> ![speedklipper2](https://user-images.githubusercontent.com/6392076/48301820-13320b00-e4f4-11e8-97ab-2cdcb7fcb6de.PNG)

I likek this the most after some time and coming back to the topic :D

**Comment by DkOuvE** [comment]
I'm trying to experiment with having the "pp" be a fast forward icon, just to throw a new idea into the mix:

![image](https://user-images.githubusercontent.com/23626715/52165080-c4130500-26f3-11e9-987c-1cef05a28bde.png)

Not sure about it now, but someone might be able to make it work...

**Comment by BKLronin** [comment]
> I'm trying to experiment with having the "pp" be a fast forward icon, just to throw a new idea into the mix:
> 
> ![image](https://user-images.githubusercontent.com/23626715/52165080-c4130500-26f3-11e9-987c-1cef05a28bde.png)
> 
> Not sure about it now, but someone might be able to make it work...

Thats a pretty good idea, would  leave it readable better than the "skip K" and has a message.

**Comment by DkOuvE** [comment]
2 more ideas on the table. 
![image](https://user-images.githubusercontent.com/23626715/52165445-c0ce4800-26f8-11e9-8a73-7e3dec643cbc.png)

**Comment by JohnEdwa** [comment]
@DruckiMcDruck I like that one though my first idea was "I wonder if it could fit recursion in it as well?" as it already is a 'K' in itself.


The text is too small for it but basically this (literally 10 minutes in msPaint) :
![klippar](https://user-images.githubusercontent.com/5417183/52165381-d8adc800-2708-11e9-8e89-793b416c91b2.png)
...but with a few more pixels.

**Comment by mtsgsd** [comment]
With apologies to @shoreliner I made the change mentioned by Kevin and I'm using it on the Klipper.info page. Hope you don't mind. :)

![KlipperLogo](https://user-images.githubusercontent.com/47236656/55895388-b1003380-5b81-11e9-9883-0207ed48012d.png)

**Comment by BKLronin** [comment]
So I will use it as well. Just needed a logo. :)

**Comment by DMCShep** [comment]
Discord is now using the circular version of that too, for consistency:

![Logo v4](https://user-images.githubusercontent.com/20283733/55901663-cbdda280-5b97-11e9-9a91-c55335afc4d5.png)

At least in the interim. Should a logo become more adopted and official please tag me and I'll make it so. 😃 

Paint.NET source file:

[Klipper Logo Paint.NET.zip](https://github.com/KevinOConnor/klipper/files/3065121/Klipper.Logo.Paint.NET.zip)

**Comment by JohnEdwa** [comment]
IMHO it's too detailed and completely breaks down once downscaled. As an example, Github uses 88px as the default, then downscales them to 44px and 32px depending on where they are used. The logo handles these very poorly:

![88px](https://user-images.githubusercontent.com/5417183/55902480-3e6c7380-5bd4-11e9-871f-df6b23fbb99b.png)    ![44px](https://user-images.githubusercontent.com/5417183/55902482-3f9da080-5bd4-11e9-9ffd-a8502ede5298.png)    ![32px](https://user-images.githubusercontent.com/5417183/55902484-40363700-5bd4-11e9-9fd7-af540f34730e.png)

**Comment by BKLronin** [comment]
Seems to be pixelbased from the program. Vector would be better for a logo. (Mine were simple screenshots for posting ease)

Agree on the scaling readability.

It seems we still have a hard time to choose a logo.

**Comment by DMCShep** [comment]
I'll +1 the downscale issue, Discord doesn't leave a lot of room for the logo at all. Though it shouldn't be hard to vector with Inkscape given it's mostly PCB-based that said, but that doesn't solve the readability aspect.

Also +1 on the hard time choosing. I think it'll take Kevin to decide.

**Comment by BKLronin** [comment]
I think its ok for discord.

Yes, it can be vectorized but the circle is already pixelated and not smooth and the logo is extremely detailed so that makes it pretty slow in any other program the logo will be used. 

Kevins turn :P

I still like this one 🗡 

![klipperlogo](https://user-images.githubusercontent.com/6392076/55905553-7ecaf200-5bd2-11e9-871a-9b5f6aa88dbd.PNG)

**Comment by mtsgsd** [comment]
Looks to much like "Lipper" to me. :)

**Comment by BKLronin** [comment]
> Looks to much like "Lipper" to me. :)

Aaand here we go again :D

**Comment by KevinOConnor** [comment]
My 2 cents - of the images I've seen, I prefer this one:

![klipperlogo](https://user-images.githubusercontent.com/6392076/48005675-4ea79080-e114-11e8-923c-3df48e870815.png)

and the similar:

![klipperlogo](https://user-images.githubusercontent.com/33714214/48003682-6a5c6800-e10f-11e8-9f76-50e09796fb67.PNG)

That said, I think I might prefer this one if it were vectorized:

![klipperlogo](https://user-images.githubusercontent.com/5417183/52165381-d8adc800-2708-11e9-8e89-793b416c91b2.png)

(I think the recursive K is a neat addition from the original at https://github.com/KevinOConnor/klipper/issues/822#issuecomment-435652769 .)

Thanks.
-Kevin

**Comment by DkOuvE** [comment]
Seems reproducible enough, and scales way better than the circle one.
![image](https://user-images.githubusercontent.com/23626715/55975270-84cbdc00-5c81-11e9-9bc0-ce5702a0cbb8.png)

**Comment by DkOuvE** [comment]
> My 2 cents - of the images I've seen, I prefer this one:
> 
> ![klipperlogo](https://user-images.githubusercontent.com/6392076/48005675-4ea79080-e114-11e8-923c-3df48e870815.png)
> 
> and the similar:
> 
> ![klipperlogo](https://user-images.githubusercontent.com/33714214/48003682-6a5c6800-e10f-11e8-9f76-50e09796fb67.PNG)
> 
> That said, I think I might prefer this one if it were vectorized:
> 
> ![klipperlogo](https://user-images.githubusercontent.com/5417183/52165381-d8adc800-2708-11e9-8e89-793b416c91b2.png)
> 
> (I think the recursive K is a neat addition from the original at [#822 (comment)](https://github.com/KevinOConnor/klipper/issues/822#issuecomment-435652769) .)
> 
> Thanks.
> -Kevin

The first logo seems reproducible enough, and scales way better than the circle one.
Seems reproducible enough, and scales way better than the circle one.

![image](https://user-images.githubusercontent.com/23626715/55975270-84cbdc00-5c81-11e9-9bc0-ce5702a0cbb8.png)

**Comment by BKLronin** [comment]
Ok cool, I´ll just attach this SVG here :) 

I think the classic K does work pretyt well as small and big logo and even T-Shirts  💯 

I tried to put the K into the Rendered perspective from above but that doens´t work to well and we  can´t use a rendered logo I would say. 

Maybe Final as SVG:
[Klipperbig.zip](https://github.com/KevinOConnor/klipper/files/3070183/Klipperbig.zip)

And the almost complete collection svg. 
[KlipperDrafts.zip](https://github.com/KevinOConnor/klipper/files/3070207/KlipperDrafts.zip)

**Comment by DMCShep** [comment]
The color schemes are worlds apart from what works on dark themed stuff like Discord (in which the red part is unreadable and the gray part is camoflauged quite well), but I at least vectorized the one to have a brass nozzle (which I started before @BKLronin responded -- no harm no foul! :)

Looks like this:

![Logo Vectorized PNG](https://user-images.githubusercontent.com/20283733/55989629-c2c20380-5c73-11e9-8d6e-8f572c579c9d.png)
[Klipper Logo Vectorized.zip](https://github.com/KevinOConnor/klipper/files/3070410/Klipper.Logo.Vectorized.zip)

Might need to tweak colors a bit, but Inkscape makes this dead easy. Not an artistically inclined guy, just know my way around "trace bitmap" and manual pixel labor heh. Font used seems to be Calibri btw.

@KevinOConnor it's your brainchild, what say you re: color scheme?

**Comment by JohnEdwa** [comment]
@DMCShep It is indeed because it literally was 10 minutes in MSPaint as a quick mockup/proof of concept for the recursion idea and calibri is the default font :p

**Comment by BKLronin** [comment]
Most logos don´t work on black but they can be easily used as a white version. 

Not a fan of the elite dangerous double nozzle variant. :P

**Comment by DrNotThatEvil** [comment]
Wait nobody suggested something that looks like clippy?

**Comment by KevinOConnor** [comment]
@BKLronin - thanks.  I haven't had a chance to look at this further, but hope to in the upcoming week.

@DMCShep - I like it.  I agree that a different color scheme could improve it, but I've no insight on what those colors should be.

-Kevin

**Comment by BKLronin** [comment]
@KevinOConnor 

I might look into a set of different color schemes. I guess the violett tone isn´t for everyone even though I perceive it as kind of "fresh".

**Comment by KevinOConnor** [comment]
@dmcshep - I came up with this for a slightly different color scheme:

![klipper3](https://user-images.githubusercontent.com/3004890/56257164-52fbb080-6099-11e9-9d7f-8965435e8bc7.png)

I've been working on putting together a web page for klipper - see: https://www.klipper3d.org/ .  I think it could really use a logo (a small version to replace the "Klipper" url at the top right of the pages, and a large one for the main home screen).

@JohnEdwa , @DMCShep , @BKLronin - can you "sign off" that you're okay with the terms described in https://github.com/KevinOConnor/klipper/blob/master/docs/developer-certificate-of-origin - if it's okay, just respond here with a "signed-off-by" tag as described at: https://github.com/KevinOConnor/klipper/blob/master/docs/CONTRIBUTING.md

Thanks everyone.
-Kevin

**Comment by DMCShep** [comment]
I think I'm doing this right, if formatting is wrong or whatever I am signing off on it, though obviously contingent on @JohnEdwa also signing off as I vectored his bitmap.

Signed-off-by: Michael Shephard @DMCShep

**Comment by BKLronin** [comment]
Signed-off-by: Thomas Herrmann <therrmann23@gmail.com>

**Comment by BKLronin** [comment]
Good to hear you are working on  a website. If its jekyll github pages based let me know I got experience on this. Made my own with it.

Some more sketches. 
![KlipperMoreLogos](https://user-images.githubusercontent.com/6392076/56275328-cd135000-6100-11e9-9cae-128bea0f16ce.PNG)

**Comment by Hywelmartin** [comment]
@BKLronin I like the second one better..

**Comment by BKLronin** [comment]
@Hywelmartin 
Me too, the first one screams 80ies and Laser too much despite it has the printing layers.

**Comment by DruckiMcDruck** [comment]
> I've been working on putting together a web page for klipper - see: https://www.klipper3d.org/ . I think it could really use a logo (a small version to replace the "Klipper" url at the top right of the pages, and a large one for the main home screen).

Bit of OT here, but i'd love to see a webpage/documentation which is generated directly from markdown. This would keep the documentation on the klipper repo and in human readable form and you could still generate a visually pleasing version of it. 

I'm thinking of using something like MKDocs, Sphinx, docsify or similar software. Jekyll might also work.

**Comment by JohnEdwa** [comment]
I really didn't think my quick paint sketch would ever go anywhere, thanks @DMCShep for making it work :)

Signed-off-by: Jonne Kuusela johnedwa@gmail.com

**Comment by DMCShep** [comment]
No sweat! I'm just glad there's now an officially selected logo. I'll put a white outline around it later today for Discord, to keep the color scheme intact and also work with light themes as well.

@DruckiMcDruck - while GitBooks based, a ton of resources previously hosted on the Discord was migrated to https://klipper.info/ -- check that out if you haven't already. It's Markdown based just like the docs in this repo, but it's supplemental to them rather than a replacement.

**Comment by DruckiMcDruck** [comment]
@DMCShep 
Yes, i know that site. While i like the approach, i'd also like to see a (sub-)site to replicate the exact docs inside the repo. So for example a site like yours with general information and introduction plus a carbon copy of the docs.

**Comment by BKLronin** [comment]
> @DMCShep - I came up with this for a slightly different color scheme:
> 
> ![klipper3](https://user-images.githubusercontent.com/3004890/56257164-52fbb080-6099-11e9-9d7f-8965435e8bc7.png)
> 
> I've been working on putting together a web page for klipper - see: https://www.klipper3d.org/ . I think it could really use a logo (a small version to replace the "Klipper" url at the top right of the pages, and a large one for the main home screen).
> 
> @JohnEdwa , @DMCShep , @BKLronin - can you "sign off" that you're okay with the terms described in https://github.com/KevinOConnor/klipper/blob/master/docs/developer-certificate-of-origin - if it's okay, just respond here with a "signed-off-by" tag as described at: https://github.com/KevinOConnor/klipper/blob/master/docs/CONTRIBUTING.md
> 
> Thanks everyone.
> -Kevin
> 
> EDIT: I meant, I came up with a remix using the color scheme similar to the one @BKLronin used.

Can we finalize that logo a bit more? I´m not a fan of that version as stated but there are a few points if we go from there that I want to suggest.

- Adjust font for >lipper< as it is way softer and doesn´t match the Size of the oroginal K at front.
- Remove the extruded plastic line form the first K otherswise its redundant to me.
- Maybe chop of the extruded plastic line on the just a little to pronounce the i dot.

Or use it as it is, but it won´t hit my T-shirt that way :D

**Comment by KevinOConnor** [comment]
FYI, https://www.klipper3d.org/ is hosted by github pages and is automatically generated from the information in https://github.com/KevinOConnor/klipper/tree/master/docs via Jekyll.  This is my first time using Jekyll - if there are suggestions then that would be great.  Everything on the site is defined from the docs/ directory on the master branch (docs/_config.yml, docs/index.md, docs/_layouts/default.html).

@BKLronin - nothing is set in stone.  If you have suggestions or improvements, that's great.  Of all the logos I've seen so far, I like your K (as seen in https://github.com/KevinOConnor/klipper/issues/822#issuecomment-482185919 ) and @dmcshep's "Klipper" with the updated colors (as seen in https://github.com/KevinOConnor/klipper/issues/822#issuecomment-483916791).  As of right now, I'd go with the second one.

-Kevin

**Comment by JohnEdwa** [comment]
@BKLronin 
>Adjust font for >lipper< as it is way softer and doesn´t match the Size of the oroginal K at front.

It's currently Calibri only because that is the default in MSpaint, there are countless other fonts that could work better for sure, and the issue is mostly because I designed the original mockup as small pixel art - at that size the letters didn't have much roundness to them, but it's very pronounced with the font when scaled up. 

>Remove the extruded plastic line form the first K otherswise its redundant to me.

Then it wouldn't be hinting at recursion or look like it's a nozzle anymore though, it would just be just a K with a funny red outline in one bit. 

>Maybe chop of the extruded plastic line on the just a little to pronounce the i dot.

But that would be underextrusion! ;P

**Comment by BKLronin** [comment]
> It's currently Calibri only because that is the default in MSpaint, there are countless other fonts that could work better for sure, and the issue is mostly because I designed the original mockup as small pixel art - at that size the letters didn't have much roundness to them, but it's very pronounced with the font when scaled up.

Thats ok, I just mentioned that we can´t leave it like that for the finals :) As you can see even the "L" is bigger than the K. Just need to harmonize it with the K. There are tons of fonts out there but the difficulty is to find the right one that expresses what you want to say or make your own :)

> Then it wouldn't be hinting at recursion or look like it's a nozzle anymore though, it would just be just a K with a funny red outline in one bit.

Yes, exactly, thats enough at that point. 

> But that would be underextrusion! ;P

Just a thin seperator so we got a dot. A line is not a dot :D

**Comment by DMCShep** [comment]
@BKLronin Can you work up the proposed changed with the SVG I Provided in [this post](https://github.com/KevinOConnor/klipper/issues/822#issuecomment-482293859)? That's not my forte at all, I would need to re-bitmap it and start all over again.

SIde note: I would like to see a "favicon" version that just has the nozzle/filament without the word "Klipper", since that would be illegible at such a small size, I can probably get a white outline around the "favicon" version for the Discord logo too, not a lot of room to work with there either.

**Comment by BKLronin** [comment]
> Can you work up the proposed changes with the SVG I Provided in this post?

Yes I can go from there with the colors provided by Kevin. 

Favicon and different versions as soon as the final version is here :)

**Comment by KevinOConnor** [comment]
Ah, I hadn't noticed the "L" being larger than the "K".  I agree it would be good to fix that.  I also agree it would be good for the stem of the K to be "softened" or for the other letters to have sharper edges.

I think it would certainly be interesting to see a version that has the start of the extrusion with a "softer" edge and/or a break in the extrusion to emphasize the dot in the "i".  I do think the corresponding changes should also be on the stem of the K as well then.

FWIW, I like the simpler fonts (like Calibri).  I suspect they'll also scale better than more intricate fonts.

Thanks,
-Kevin

**Comment by theopensourcerer** [comment]
FWIW I rather like the Ubuntu font which is free and open source:

![klipper-in-ubuntu](https://user-images.githubusercontent.com/3414988/56314329-b2ff5f00-614c-11e9-9d56-848b4a798eeb.png)

**Comment by BKLronin** [comment]
@kevinoconnor

Going to implement the suggested changes. 

I don't even have calibri but I will find an open source one tomorrow like the one above. The previous fonts I used previously were all os fonts or at least an open license.

Thanks for the suggestion @theopensourcerer

**Comment by JohnEdwa** [comment]
There is an open license version very close to Calibri called [Carlito](https://fontlibrary.org/en/font/carlito) which is basically [Lato](https://fontlibrary.org/en/font/lato) but with the same letter spacing/sizing as Calibri to perfectly match the layout it in print.

But, I leave looking for the more professional fonts to other people as I tend to always gravitate towards sci-fi looking fonts ([*cough*](http://johnedwa.arkku.net/img/newlogo2.png)), but as skimmed some of the interesting looking ones from Google Fonts, I might as well post them here as my suggestions.

![aldrich](https://user-images.githubusercontent.com/5417183/56321625-cbcd3c00-616f-11e9-8e54-825691d36edd.PNG)
[Aldrich](https://fonts.google.com/specimen/Aldrich)

![Quantico](https://user-images.githubusercontent.com/5417183/56321632-cff95980-616f-11e9-8d32-9de3854ad812.PNG)
[Quantico](https://fonts.google.com/specimen/Quantico)

![Rationale](https://user-images.githubusercontent.com/5417183/56321638-d2f44a00-616f-11e9-9655-f60555afcd9c.PNG)
[Rationale](https://fonts.google.com/specimen/Rationale)

![Geo](https://user-images.githubusercontent.com/5417183/56321641-d5ef3a80-616f-11e9-9593-1351e9ce0278.PNG)
[Geo](https://fonts.google.com/specimen/Geo)

![Chathura](https://user-images.githubusercontent.com/5417183/56321649-d982c180-616f-11e9-9906-6504cc28f959.PNG)
[Chathura](https://fonts.google.com/specimen/Chathura)

**Comment by BKLronin** [comment]
@JohnEdwa 

A minute too late but I like scififonts as well. Lets have a look at the Ubuntu Version first. I corrected
[KlipperRC1704.zip](https://github.com/KevinOConnor/klipper/files/3091626/KlipperRC1704.zip)

 some smaller defects. I saw the logo as an icon on the desktop  and liked it suddenly :D

![KlipperRC_prev](https://user-images.githubusercontent.com/6392076/56322226-c15b7400-6168-11e9-944c-f059b548357f.png)

**Comment by KevinOConnor** [comment]
Thanks.

FWIW, I'm not a fan of the flourishes on the "L".

I made some tweaks and came up with the following:

![klipper4 svg](https://user-images.githubusercontent.com/3004890/56328924-3e79ef80-614f-11e9-9ad2-e82cc0a57985.png)

[klipper4.zip](https://github.com/KevinOConnor/klipper/files/3092012/klipper4.zip)

-Kevin

**Comment by DMCShep** [comment]
I'm not sure when this started -- it might very well have been me -- but I just spotted there's a gap in some places that probably shouldn't be there. Take a look:

![Klipper5 Gap](https://user-images.githubusercontent.com/20283733/56332851-ebf4ff00-615f-11e9-8210-f2e62b684e8d.png)

This is also true of the "K" in Klipper, which has the same geometry.

Also, I noticed the tops and bottoms of the "Kli" don't line up fully -- and I don't know if the "r" lines up either on the bottom (or of it needs to either).

![Klipper5 Baseline Zoom](https://user-images.githubusercontent.com/20283733/56333470-4db66880-6162-11e9-8365-ce0ac063b105.png)

Close-up of spot # 1 (bottoms) with a line aligned with the bottom of the "K" for reference:

![Klipper5 Baseline](https://user-images.githubusercontent.com/20283733/56333499-6757b000-6162-11e9-95d8-e24ebc16c21c.png)

The L is a tad high and the i is way high. Easy fix for someone that knows their way around an SVG, not for me though.

**Comment by mtsgsd** [comment]
> > I've been working on putting together a web page for klipper - see: https://www.klipper3d.org/ . I think it could really use a logo (a small version to replace the "Klipper" url at the top right of the pages, and a large one for the main home screen).
> 
> Bit of OT here, but i'd love to see a webpage/documentation which is generated directly from markdown. This would keep the documentation on the klipper repo and in human readable form and you could still generate a visually pleasing version of it.
> 
> I'm thinking of using something like MKDocs, Sphinx, docsify or similar software. Jekyll might also work.
Our klipper.info page is meant to be a resource for people trying to learn how to install and use it on various hardware, while not trying to be a copy of the documentation. I refer everyone to the /docs on the github, but I'll be happy to change it to point to whatever source Kevin chooses. I want to link to the real thing rather than try to copy it all and then keep it synchronized.

**Comment by KevinOConnor** [comment]
I think I got the alignment issues fixed:  
![klipper6 svg](https://user-images.githubusercontent.com/3004890/56339911-1a81d280-617e-11e9-87a3-75d856e5ca13.png)

[klipper6.zip](https://github.com/KevinOConnor/klipper/files/3092961/klipper6.zip)

-Kevin

**Comment by Hywelmartin** [comment]
There's something that's of, probably an illusion. But the side where the filament is layed down the nozzle seems lower..  
And my opinion is that the K in the word should be a normal letter...it's too much I think as it is.

**Comment by BKLronin** [comment]
@KevinOConnor Looking good, I actually liked the flouriches on th "l" and "i" as the font uses this the waqy through now these two seem a little seperated by simplicity. 

The nozzleends have the exact same height I checked it @Hywelmartin 

Sorry for the K alignment it was late :D

I wasn´t sure with the second nozzle K at the beginning but now I think it works as a good anchor point for the I to dive into the logo and explains the V better.

Plot Twist ! 
![Stresstest](https://user-images.githubusercontent.com/6392076/56348569-03b39e00-61c7-11e9-9e8d-4e6bcfbb9c74.png)

**Comment by Hywelmartin** [comment]
Not to be nitpicking but.. the K and the "V" is not the same. 
The lines are not parallell, they have different spacing,  the filament got an fillet in the "V" and a spline on the K.
I like the spline better on the filament in the K
![image](https://user-images.githubusercontent.com/33714214/56357194-4aab8e80-61db-11e9-864b-5540fa746280.png)
![image](https://user-images.githubusercontent.com/33714214/56357222-62831280-61db-11e9-92f8-2b4709997ca4.png)
![image](https://user-images.githubusercontent.com/33714214/56357663-9a3e8a00-61dc-11e9-8cbd-5ed133843625.png)

**Comment by BKLronin** [comment]
@Hywelmartin 
I corrected that filet on both the K and V yesterday manually as they weren´t round. They are both splines and I had to manually adjust the knots so it most likely wasn´t perfect. 

The missing uniformity of the K and V most likely result from the different iterations and conversions and the idea sketching. Should be fine as it´s not a technical drawing allthough it should be as precise as possible.

Neat idea with the K line as heated bed but I don´t like the altered color tones if that was intended.

**Comment by DkOuvE** [comment]
Throwing some thowback ideas into the fire:
![image](https://user-images.githubusercontent.com/23626715/56361517-7e88b300-61df-11e9-995b-0b0d2fa38334.png)

**Comment by JohnEdwa** [comment]
Also with vectors, if you place them next to each other you will almost always get issues from the anti-aliasing creating semi-transparent gaps, as happens between the red and blue parts:

![gappy](https://user-images.githubusercontent.com/5417183/56363683-75a2dc00-61f5-11e9-88c0-d1fe820f4e5a.png)

They should ever so slightly overlap or have an additional line under both of them that bridges the gap.

**Comment by KevinOConnor** [comment]
Thanks.  It's getting hard to track all the revisions here, so I committed the logo to the repo (commit 9b960e63).  I also fixed up some of the anti-aliasing (commit 49b18a6b).

https://github.com/KevinOConnor/klipper/blob/master/docs/img/klipper.svg

-Kevin

**Comment by Hywelmartin** [comment]
@BKLronin  don't mind the color.. I did just ad some similar colours for show..
 
![image](https://user-images.githubusercontent.com/33714214/56374575-c9b4bd00-6203-11e9-9c85-f3163dde46bc.png)

**Comment by BKLronin** [comment]
@Hywelmartin 
Thats cool, I mostly thought of side patterns but missed on the more interesting top surfaces. MIght play with the appearance of the lines but I guess it´s to detailed for a logo.

**Comment by Hywelmartin** [comment]
this is straight out from the slicer

**Comment by BKLronin** [comment]
Thought so but wasn´t sure if you converted it into something.

**Comment by Hywelmartin** [comment]
![image](https://user-images.githubusercontent.com/33714214/56380235-f243b380-6211-11e9-86c3-0ee4b1e262f9.png)

**Comment by BKLronin** [comment]
Added as draft into that version. 

[Klipperretro.zip](https://github.com/KevinOConnor/klipper/files/3095650/Klipperretro.zip)

**Comment by DMCShep** [comment]
Given this ticket has hit a bit of a lull, I tried to put [the logo in master](https://github.com/KevinOConnor/klipper/blob/master/docs/img/klipper.svg) onto Discord with a white outline. It did not go over well and is far too busy with the white outline in combination with the letters and logo all in one tiny format. It was a lot all at once, looked busier than the circle circuitboard logo we're using now. We need a dark theme color scheme to compliment the current coloring. The red is actually not bad on its own, but the blue blends in with the Discord theme too much to leave as-is.

As an aside: someone mentioned merch above, it would make sense to have a design that works well on black T shirts / stickers / printers as well as one that works on white T shirts / stickers / printers, so this has other applications as well.

**Comment by BKLronin** [comment]
Just turn it into white :) 

Even the amazon logo doesn't work on black.

**Comment by Elocai** [comment]
V because V stands for Vlipper.... why?
joking aside I don´t feel a V represents Klipper well could we Turn it 90° And just write "K" Klipper as | top to buttom or turned?

**Comment by BKLronin** [comment]
> Couldn´t resist as well :D
> ![merchandising](https://user-images.githubusercontent.com/6392076/47935987-9b088b80-dedc-11e8-9c19-ab1c92944322.PNG)

Like this? Sadly that version got rejected :(

Just imagine that the V stands for velocity :)

**Comment by JohnEdwa** [comment]
@Elocai, In case you really didn't get it, it's a 3D printer nozzle printing on the 'bed', a rotated "K" and it has recursion as more clearly illustrated in my previous comment. The word "Klipper" is part of the logo, it's not just the "V".

I prefer mine (obviously), but if the logo would be rotated, I feel the design @BLKRonin has would be better as it has a much stronger "K", while both designs pretty much completely lose the idea of a nozzle. It's much easier to see the "rotated K for Klipper" than the "Oh it's a nozzle but sideways how clever", at least to me.

And as the most important parts is how they look small, here they are at 40x40px: 
![Untitled-2](https://user-images.githubusercontent.com/5417183/56609321-797b9780-6615-11e9-8db9-679de4e9fb4d.png)  ![asd](https://user-images.githubusercontent.com/5417183/56610375-f60f7580-6617-11e9-8fd2-bf45864009f1.png) ![Untitled-1](https://user-images.githubusercontent.com/5417183/56609285-62d54080-6615-11e9-81df-9ff7c0b3c14f.png) 


@DMCShep, Obviously any logo with flat colour will have backgrounds it won't work with, which is why once the design is chosen and finalized there needs to be colour/design variations for them.
Twitter for example has three: 
![twiutlogo](https://user-images.githubusercontent.com/5417183/56608723-1b9a8000-6614-11e9-88b9-b6fcc0b000c1.PNG)

**Comment by BKLronin** [comment]
The 90 degree turned masterlogo looks interesting..

I instantly had the impression of someone stepping out from behind a wall.

**Comment by DMCShep** [comment]
I wasn't sure if a lighter tone to the blue could be had, though leaving the red the original color doesn't look half bad either:

![image](https://user-images.githubusercontent.com/20283733/56596287-4954da00-65be-11e9-82ec-aa6335ae56c5.png)

Discord's background color is #202225 for reference. For now I'll go with the red "filament" until and unless something more suitable comes along.

**Comment by BKLronin** [comment]
> Just turn it into white :)
> 
> Even the amazon logo doesn't work on black.
> [KlipperB&W.zip](https://github.com/KevinOConnor/klipper/files/3108296/KlipperB.W.zip)

Have you seen this version? White and light grey, as the red white black looked a bit brutal to me.

**Comment by DMCShep** [comment]
The light gray becomes nearly invisible against the Discord background in that version:

![image](https://user-images.githubusercontent.com/20283733/56620506-1ed34300-65f7-11e9-8e63-1979aa3a78e4.png)

I would love something that looked better on the Discord background but it's not playing nice at the moment.

**Comment by BKLronin** [comment]
I see,
![klipperwhitegrey](https://user-images.githubusercontent.com/6392076/56621005-214f9000-662b-11e9-8cc9-6ded4daab5ef.PNG)
![klipperwhitegrey90](https://user-images.githubusercontent.com/6392076/56621006-214f9000-662b-11e9-969d-c04273e56be3.PNG)

**Comment by BKLronin** [comment]
[KlipperB&W_lighter.zip](https://github.com/KevinOConnor/klipper/files/3109875/KlipperB.W_lighter.zip)

**Comment by DMCShep** [comment]
I don't know that black & white is the best choice for dark theme though. Case in point, if I change the blue to H:210 S:30 L:50 (#597FA5), I get this, which is less harsh than red and white and is still pleasing enough:

![image](https://user-images.githubusercontent.com/20283733/56624539-534efb00-6607-11e9-9b76-1a2060d23194.png)

Compared to the original:

![image](https://user-images.githubusercontent.com/20283733/56624626-a0cb6800-6607-11e9-9e0d-adfa2fddf25a.png)

This is an improvement in that blue is more visible, but I still feel the blue/red balance needs to be tweaked more to make them mesh well together.

**Comment by BKLronin** [comment]
Not too bad, I think the darker grey looks more valueable. If we change colors within the logo we should keep that on the noramle white background as well for consistency. I made some color variants that are technically matching the red.

![KLColorvariations](https://user-images.githubusercontent.com/6392076/56644070-41a73b00-667b-11e9-9605-7e900f1190d0.png)

**Comment by Minasokoni** [comment]
![image](https://user-images.githubusercontent.com/1010168/56682956-7918e600-6681-11e9-8261-bdccbf614403.png)
![image](https://user-images.githubusercontent.com/1010168/56683266-02301d00-6682-11e9-9b45-461d4f8211ac.png)

**Comment by BKLronin** [comment]
The first one is great. The other seems a bit develish.

**Comment by Hywelmartin** [comment]
@Minasokoni nice one. Best so far in my opinion.. simple, recognisable, machinable...
@BKLronin I like the devilishness... 
![image](https://user-images.githubusercontent.com/33714214/56726156-89e35d80-674e-11e9-9002-e27a5825a868.png)
![image](https://user-images.githubusercontent.com/33714214/56726284-c0b97380-674e-11e9-8c50-d4cb9a2ed851.png)
![image](https://user-images.githubusercontent.com/33714214/56726361-daf35180-674e-11e9-8006-4d13ccf35f0f.png)

**Comment by Minasokoni** [comment]
@Hywelmartin thank you.
I'm removing the 2nd logo

**Comment by BKLronin** [comment]
@KevinOConnor 
Look Kevin it´s perfect, might consider this one. https://github.com/KevinOConnor/klipper/issues/822#issuecomment-486366179

**Comment by Minasokoni** [comment]
@BKLronin @KevinOConnor I can also help out with the website as well

**Comment by Hywelmartin** [comment]
I think that we keep the devilish on as well

**Comment by BKLronin** [comment]
I guess its better to keep thungs simple and not introduce too much variants or this will be an endless story. It should be the logo we got and bw variants vs the new challenger and let Kevin decide. 

The aquamarin logo looks pretty solid and professionnal. 

Help on the website would be helpful  just can make clean looking sites with html and bootstrap on jekyll.

**Comment by JohnEdwa** [comment]
While it would be rather nifty as a standalone logo, the combination of the squiggly and two-tone colouring means it's really had to get the "K" from it, it just looks like it says "lipper" or "<lipper". It also kinda reminds me of a snake, which being green-ish doesn't help at all, i think blue or red would be much better. [Yellow is right out though.](https://images-na.ssl-images-amazon.com/images/I/61MtBT9XDdL._SX425_.jpg)

But, with that design, I feel that the logo and name would need to be separated, something like these :
![browsera](https://digitalsynopsis.com/wp-content/uploads/2017/06/creative-single-letter-logo-designs-browsera.jpg)
![elefont](https://digitalsynopsis.com/wp-content/uploads/2017/06/creative-single-letter-logo-designs-elefont.jpg)

**Comment by KevinOConnor** [comment]
Thanks everyone.  Of all the logos I've seen, my favourites are the one currently in use (https://www.klipper3d.org/img/klipper-logo.png) and https://user-images.githubusercontent.com/6392076/48005675-4ea79080-e114-11e8-923c-3df48e870815.png .  For a logo with a dark background, my initial guess is that the one at the top right of https://github.com/KevinOConnor/klipper/issues/822#issuecomment-486121971 looks good (though it's hard to tell with the white background of github).

The image at https://github.com/KevinOConnor/klipper/issues/822#issuecomment-486366179 is neat, but I fear the symbol at the start doesn't look enough like a K.

If there are suggestions on the website, feel free to create a github pull request, or list out the changes that you think would make sense (might be good to open a new issue).

Thanks again,
-Kevin

**Comment by KevinOConnor** [comment]
It looks like the conversation has concluded on this topic.  I'm going to close this issue.  A new issue can be opened if future enhancements are desired.  Thanks everyone for their contributions.

-Kevin
